File: docs/snippets/all/archetypes/grid_map_simple.rs
//! Log a simple occupancy grid map.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_grid_map").spawn()?;

    let width: usize = 64;
    let height: usize = 64;
    let cell_size: f32 = 0.1;

    // Create a synthetic image with ROS `nav_msgs/OccupancyGrid` cell value conventions:
    // -1 (255) unknown, 0 free, 100 occupied.
    let mut grid = vec![255u8; width * height];
    for y in 8..56 {
        for x in 8..56 {
            grid[y * width + x] = 0;
        }
    }
    for y in 20..44 {
        for x in 20..44 {
            grid[y * width + x] = 100;
        }
    }

    rec.log(
        "world/map",
        &rerun::GridMap::new(
            grid,
            rerun::components::ImageFormat::from_color_model(
                [width as u32, height as u32],
                rerun::ColorModel::L,
                rerun::ChannelDatatype::U8,
            ),
            cell_size,
        )
        .with_translation([
            -(width as f32) * cell_size / 2.0,
            -(height as f32) * cell_size / 2.0,
            0.0,
        ])
        .with_colormap(rerun::components::Colormap::RvizMap),
    )?;

    Ok(())
}

File: docs/snippets/all/archetypes/grid_map_simple.py
"""Log a simple occupancy grid map."""

import numpy as np

import rerun as rr

width, height = 64, 64
cell_size = 0.1

# Create a synthetic image with ROS `nav_msgs/OccupancyGrid` cell value conventions:
# -1 (255) unknown, 0 free, 100 occupied.
grid = np.full((height, width), -1, dtype=np.int8)
grid[8:56, 8:56] = 0
grid[20:44, 20:44] = 100

rr.init("rerun_example_grid_map", spawn=True)

rr.log(
    "world/map",
    rr.GridMap(
        data=grid.tobytes(),
        format=rr.components.ImageFormat(
            width=width,
            height=height,
            color_model="L",
            channel_datatype="U8",
        ),
        cell_size=cell_size,
        translation=[-(width * cell_size) / 2.0, -(height * cell_size) / 2.0, 0.0],
        colormap=rr.components.Colormap.RvizMap,
    ),
)
